airbyte-cdk/python/airbyte_cdk/connector_builder/connector_builder_handler.py

```diff
@@ -105,20 +113,20 @@ def list_streams(source: ManifestDeclarativeSource, config: Dict[str, Any]) -> A
         return AirbyteTracedException.from_exception(exc, message=f"Error listing streams: {str(exc)}").as_airbyte_message()
 
 
-def _get_http_streams(source: ManifestDeclarativeSource, config: Dict[str, Any]) -> List[HttpStream]:
+def _get_http_streams(source: ManifestDeclarativeSource, config: Dict[str, Any]) -> List[SimpleRetriever]:
     http_streams = []
     for stream in source.streams(config=config):
         if isinstance(stream, DeclarativeStream):
-            if isinstance(stream.retriever, HttpStream):
+            if isinstance(stream.retriever, SimpleRetriever):
```

Comment:
That is what happens in practice, but it's not required - every requester has `get_url_base` and `get_path` which is what we are using here.

We need to know it's a `SimpleRetriever`, because the fact that a retriever has a `requester` is an implementation detail of `SimpleRetriever` but we can work on the general `Requester` interface for the requester itself.